src/devices/GrovePi/Sensors/ISensor.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Iot.Device.GrovePiDevice.Models;
+using System.Collections.Generic;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+	/// Interface for a sensor 
+	/// </summary>
+    public interface ISensor<T>
```

Comment:
This is interesting idea to represent all sensors. I think we should also have specific types which use that as well. So like in my PR: https://github.com/dotnet/iot/pull/256 I presume ITemperatureSensor could also imply `ISensor<float>`? But currently that would also imply that we always return "Temperature sensor" in the SensorName field or should this be specific device name? If we go with more generic then interface can't do that in C# (AFAIK)

Reply:
Absolutely. The temperature sensor should implement ISensor<float>. I like to have at least a standard way to get access to the "raw" data of the sensor like Value.  And for the sensor name. Makes it easy to display. 
Now, yes, it's a bit broader discussion with the types as well. And yes, some sensors can have multiple interfaces like a temperature one, humidity one and we can think of few others. And I'll catch up in the other PR.